stdlib/universe/chande_momentum_oscillator.go

```diff
@@ -0,0 +1,295 @@
+	return &chandeMomentumOscillatorTransformation{
+		d:       d,
+		cache:   cache,
+		n:       spec.N,
+		columns: spec.Columns,
+	}
+}
+
+func (t *chandeMomentumOscillatorTransformation) RetractTable(id execute.DatasetID, key flux.GroupKey) error {
+	return t.d.RetractTable(key)
+}
+
+func (t *chandeMomentumOscillatorTransformation) Process(id execute.DatasetID, tbl flux.Table) error {
+	builder, created := t.cache.TableBuilder(tbl.Key())
+	if !created {
+		return fmt.Errorf("chande momentum oscillator found duplicate table with key: %v", tbl.Key())
+	}
+	cols := tbl.Cols()
+	doChandeMomentumOscillator := make([]bool, len(cols))
+	for j, c := range cols {
+		found := false
+		for _, label := range t.columns {
+			if c.Label == label {
+				if c.Type != flux.TInt && c.Type != flux.TUInt && c.Type != flux.TFloat {
+					return fmt.Errorf("cannot take chande momentum oscillator of column %s (type %s)", c.Label, c.Type.String())
+				}
+				found = true
+				break
+			}
+		}
+
+		if found {
+			mac := c
+			mac.Type = flux.TFloat
+			_, err := builder.AddCol(mac)
+			if err != nil {
+				return err
+			}
+			doChandeMomentumOscillator[j] = true
+		} else {
+			_, err := builder.AddCol(c)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return tbl.Do(func(cr flux.ColReader) error {
+		if cr.Len() == 0 || len(cr.Cols()) == 0 {
+			return nil
+		}
+
+		for j, c := range cr.Cols() {
+			if !doChandeMomentumOscillator[j] {
+				for i := int(t.n); i < cr.Len(); i++ {
+					if err := builder.AppendValue(j, execute.ValueForRow(cr, i, j)); err != nil {
+						return err
+					}
+				}
+				continue
+			}
+			if err := t.do(int(t.n), cr, c, builder, j); err != nil {
+				return err
+			}
+		}
+		return nil
+	})
+}
+
+func (t *chandeMomentumOscillatorTransformation) do(n int, cr flux.ColReader, c flux.ColMeta, builder execute.TableBuilder, j int) error {
+	var sumUp float64
+	var sumDown float64
+	sumUp = 0
+	sumDown = 0
+
+	switch c.Type {
+	case flux.TInt:
+		arrValues := cr.Ints(j)
+		c.Type = flux.TFloat
+		prev := arrValues.Value(0)
+		curr := int64(0)
+		for i := 0; i < cr.Len(); i++ {
+			curr = arrValues.Value(i)
+			diff := float64(curr - prev)
+			if i >= n {
+				diffNAgo := float64(arrValues.Value(i-n+1) - arrValues.Value(i-n))
+				val, su, sd := nextCMO(sumUp, sumDown, diff, diffNAgo)
+				sumUp = su
+				sumDown = sd
+				if err := builder.AppendFloat(j, val); err != nil {
+					return err
+				}
+			} else {
+				_, sumUp, sumDown = nextCMO(sumUp, sumDown, diff, 0)
+			}
+			prev = curr
+		}
+	case flux.TUInt:
+		arrValues := cr.UInts(j)
+		c.Type = flux.TFloat
+		prev := arrValues.Value(0)
+		curr := uint64(0)
+		for i := 0; i < cr.Len(); i++ {
+			curr = arrValues.Value(i)
+			diff := float64(curr - prev)
+			if i >= n {
+				diffNAgo := float64(arrValues.Value(i-n+1) - arrValues.Value(i-n))
+				val, su, sd := nextCMO(sumUp, sumDown, diff, diffNAgo)
```

Comment:
nit, you can just write this as val, sumUp, sumDown = nextCMO

probably you got an error about val.   Just declare it along with sumUp and sumDown at the beginning of this function.